File: web/src/types/quest.ts
export interface Achievement {
  achievement_id: string
  name: string
  description: string
  icon: string
  exp_reward: number
  unlocked: boolean
  unlocked_at: string | null
}

export interface QuestSummary {
  level: number
  exp: number
  exp_to_next: number
  streak: number
  today_total: number
  today_completed: number
  tasks_completed_total: number
  achievements_completed: number
  achievements_total: number
  achievements: Achievement[]
}

File: server/app/schemas/quest.py
from datetime import datetime

from pydantic import BaseModel


class AchievementResponse(BaseModel):
    achievement_id: str
    name: str
    description: str
    icon: str
    exp_reward: int
    unlocked: bool
    unlocked_at: datetime | None = None


class QuestSummaryResponse(BaseModel):
    level: int
    exp: int
    exp_to_next: int
    streak: int
    today_total: int
    today_completed: int
    tasks_completed_total: int
    achievements_completed: int
    achievements_total: int
    achievements: list[AchievementResponse]
